src/schema/resolvers/signup_resolver.js

```diff
@@ -0,0 +1,30 @@
+
+import models from '../../models'
+import createToken from '../../auth'
+
+const resolvers = {
+  Mutation: {
+    async signup (parent, { data }, context) {
+      const { username, password, lastName, firstName } = data
+      if (!username || !password) return { authError: 'No password or username' }
+      const count = await models.user.count({ where: { username: username } })
+      if (count > 0) {
+        return { authError: 'The username already exists' }
+      } else {
+        const user = await models.user.create({ username, password, lastName, firstName })
+        return {
+          user: user,
+          jwt: createToken(userInformation(user))
+
+        }
+      }
+    }
+  }
+}
+
+function userInformation (user) {
```

Comment:
I'd add the `get` word to this function: `getUserInformation`. Yet, I don't like it very much, it's ambiguous. Take a look at this article, it may help you with naming: https://dmitripavlutin.com/coding-like-shakespeare-practical-function-naming-conventions.

Also, this object contains one only key `information` and its value is user's username. I think that's not what you want.

You should return something like this:
```
return {
  username: user.username,
  firstName: user.firstName,
  ...etc
}
```